src/v3/src/components/LoopbackProbe/LoopbackProbe.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+// Allow this shadow dependency from auth-js because it is temporary
+// eslint-disable-next-line import/no-extraneous-dependencies
+import crossFetch from 'cross-fetch';
+import { FunctionComponent } from 'preact';
+import { useEffect } from 'preact/hooks';
+
+import Logger from '../../../../util/Logger';
+import { useOnSubmit } from '../../hooks';
+import { LoopbackProbeElement } from '../../types';
+import { isAndroid } from '../../util';
+
+type RequestOptions = {
+  url: string;
+  timeout: number;
+  method: 'GET' | 'POST';
+  data?: string;
+};
+
+/**
+ * Temporary request client to be removed once this functionality is added
+ * to auth-js library.
+ * @see https://oktainc.atlassian.net/browse/OKTA-561852
+ * @returns Promise<Response>
+ */
+const makeRequest = async ({
+  url, timeout, method, data,
+}: RequestOptions) => {
+  const controller = new AbortController();
+  const id = setTimeout(() => controller.abort(), timeout);
+  const fetch = global.fetch || crossFetch;
+
+  const response = await fetch(url, {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    ...(typeof data === 'string' ? { body: data } : {}),
+    signal: controller.signal,
+  });
+
+  clearTimeout(id);
+
+  return response;
+};
+
+const LoopbackProbe: FunctionComponent<{ uischema: LoopbackProbeElement }> = ({
+  uischema: {
+    options: {
+      deviceChallengePayload,
+      cancelStep,
+      step,
+    },
+  },
+}) => {
+  const onSubmitHandler = useOnSubmit();
+
+  const probeTimeoutMillis: number = typeof deviceChallengePayload.probeTimeoutMillis === 'undefined'
+    ? 100 : deviceChallengePayload.probeTimeoutMillis;
+  const ports: number[] = deviceChallengePayload.ports || [];
+  const { domain } = deviceChallengePayload;
+  const { challengeRequest } = deviceChallengePayload;
+
+  const cancelHandler = (params?: Record<string, unknown> | undefined) => {
```

Comment:
Just curious, is using the `ActionParams` type possible here? 
`const cancelHandler = (params?: ActionParams | undefined) => {`

Also, is `undefined` necessary if the parameter is optional?

Reply:
Good point, I think the `undefined` part was an artifact of a previous implementation. And I didn't know about the `ActionParams` type so that works great, thanks!